README.md

```diff
@@ -142,24 +147,42 @@ implementation.
 
 ## What Can I Use Serena For?
 
+Serena is a toolkit, which means it can be used in various ways, depending on your needs and preferences.
+It can be used as a highly configurable MCP server (application), as a toolkit for building agents (library), or 
+as a framework for building custom integrations based on language servers (the solidlsp package).
+
+### As MCP Server
 You can use Serena for any coding tasks – whether it is focussed on analysis, planning, 
 designing new components or refactoring existing ones.
 Since Serena's tools allow an LLM to close the cognitive perception-action loop, 
 agents based on Serena can autonomously carry out coding tasks from start to finish – 
-from the initial analysis to the implementation, testing and, finally, the version
-control system commit.
+from the initial analysis to the implementation, testing and, finally, the commit.
 
 Serena can read, write and execute code, read logs and the terminal output.
 While we do not necessarily encourage it, "vibe coding" is certainly possible, and if you 
 want to almost feel like "the code no longer exists",
 you may find Serena even more adequate for vibing than an agent inside an IDE
 (since you will have a separate GUI that really lets you forget).
 
+The application layer of Serena is highly configurable, so you can adapt it to your needs.
+You can select the tools you want to use, adjust the prompts and even the tool descriptions
+without writing any code. Serena comes with a CLI to facilitate this.
+
+See the sections on [configuration](#configuration) and [modes and contexts](#modes-and-contexts) 
+for more information on how to adapt Serena to your needs,
+and explore the cli with `uvx --from git+https://github.com/oraios/serena serena --help`.
+
+### As Library
+While the lower level interfaces should not be considered as stable yet and have no extensive documentation,
+we do develop them with generality and ease of use in mind. 
+As a result, you can use Serena as a library to build your own agents using the provided tools
+and language server capabilities.
```

Comment:
This section should be very short and to the point; without any sub-headings. Adding so much information here is counter-productive and conflates usage modes (how?) with the purpose (what?). This section is supposed to be about the latter only (as the title suggests).

Reply:
It would also refrain from trying to mention "everything" "everywhere". We definitely don't need links to configuration or modes and contexts here, for example. This is overloaded.